state/src/account.rs

```diff
@@ -0,0 +1,406 @@
+            code_hash: account.code_hash,
+            code: vec![],
+            code_size: 0,
+            code_state: CodeState::Clean,
+            storage_changes: HashMap::new(),
+        }
+    }
+}
+
+impl StateObject {
+    /// Create a new account.
+    /// NOTE: If contract account generated, make sure you use `init_code` on
+    /// this before `commit`ing.
+    pub fn new(balance: U256, nonce: U256) -> StateObject {
+        StateObject {
+            balance,
+            nonce,
+            storage_root: SHA3_NULL_RLP,
+            code_hash: SHA3_EMPTY,
+            code: vec![],
+            code_size: 0,
+            code_state: CodeState::Clean,
+            storage_changes: HashMap::new(),
+        }
+    }
+
+    /// Create a new account from rlp bytes.
+    /// Note: make sure you use `read_code` after this.
+    pub fn from_rlp(data: &[u8]) -> Result<StateObject, Error> {
+        let account: Account = rlp::decode(data).or(Err(Error::InvalidRLP))?;
+        Ok(account.into())
+    }
+
+    pub fn rlp(&self) -> Vec<u8> {
+        rlp::encode(&self.account())
+    }
+
+    pub fn account(&self) -> Account {
+        Account {
+            balance: self.balance,
+            nonce: self.nonce,
+            storage_root: self.storage_root,
+            code_hash: self.code_hash,
+        }
+    }
+
+    pub fn init_code(&mut self, code: Vec<u8>) {
+        self.code = code.clone();
+        self.code_hash = From::from(&Sha3_256::digest(&code)[..]);
+        self.code_size = code.len();
+        self.code_state = CodeState::Dirty;
+    }
+
+    pub fn read_code<B: DB>(&mut self, db: &mut B) -> Result<Vec<u8>, Error> {
+        if self.code_hash == SHA3_EMPTY {
+            return Ok(vec![]);
+        }
+        if !self.code.is_empty() {
+            return Ok(self.code.clone());
+        }
+        let c = db
+            .get(&self.code_hash)
+            .or(Err(Error::DBError))?
+            .unwrap_or_else(|| vec![]);
+        self.code = c.clone();
+        self.code_size = c.len();
+        self.code_state = CodeState::Clean;
+        Ok(c)
+    }
+
+    pub fn balance(&self) -> U256 {
+        self.balance
+    }
+
+    pub fn nonce(&self) -> U256 {
+        self.nonce
+    }
+
+    pub fn code(&self) -> Option<Vec<u8>> {
+        if self.code.is_empty() {
+            return None;
+        }
+        Some(self.code.clone())
+    }
+
+    pub fn code_hash(&self) -> H256 {
+        self.code_hash
+    }
+
+    pub fn code_size(&self) -> usize {
+        self.code_size
+    }
+
+    pub fn inc_nonce(&mut self) {
+        self.nonce += U256::from(1u8);
+    }
+
+    pub fn add_balance(&mut self, x: U256) {
+        let (a, b) = self.balance.overflowing_add(x);
+        assert_eq!(b, false);
+        self.balance = a;
+    }
+
+    pub fn sub_balance(&mut self, x: U256) {
+        let (a, b) = self.balance.overflowing_sub(x);
+        assert_eq!(b, false);
+        self.balance = a;
```

Comment:
`assert_eq!(b, false)` will be panic....

Reply:
Best comment on why here is panic.

Reply:
Assert in code means a situation that should never happend in any case.

Reply:
I think it's no problem here. The balance should never overflow in EVM.